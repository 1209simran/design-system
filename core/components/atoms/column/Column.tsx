import * as React from 'react';
import classNames from 'classnames';

type columns = '1' | '2' | '3' | '4' | '5' | '6' | '7' | '8' | '9' | '10' | '11' | '12' | 'auto';

interface IProps {
  size?: columns;
  sizeS?: columns | 's';
  sizeM?: columns | 'm';
  sizeL?: columns | 'l';
  sizeXL?: columns | 'xl';
}

const Column: React.FunctionComponent<IProps> = props => {
  const { size, sizeS, sizeM, sizeL, sizeXL } = props;

  const classes = classNames({
    ['Col']: !size,
    ['Col--s']: sizeS && sizeS === 's',
    ['Col--m']: sizeM && sizeM === 'm',
    ['Col--l']: sizeL && sizeL === 'l',
    ['Col--xl']: sizeXL && sizeXL === 'xl',
    [`Col--${size}`]: size,
    [`Col--s-${sizeS}`]: sizeS && sizeS !== 's',
    [`Col--m-${sizeM}`]: sizeM && sizeM !== 'm',
    [`Col--l-${sizeL}`]: sizeL && sizeL !== 'l',
    [`Col--xl-${sizeXL}`]: sizeXL && sizeXL !== 'xl'
  });
  return <div className={classes}>{props.children}</div>;
};

export default Column;
