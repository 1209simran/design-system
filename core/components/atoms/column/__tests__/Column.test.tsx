import * as React from 'react';
import { shallow } from 'enzyme';
import Column from '../Column';

describe('Column component', () => {
  it('should match the snapshot without columns', () => {
    const tree = shallow(<Column>Primary</Column>);
    expect(tree).toMatchSnapshot();
  });

  it('should match the snapshot with columns', () => {
    const tree = shallow(<Column size="12">Primary</Column>);
    expect(tree).toMatchSnapshot();
  });
});
