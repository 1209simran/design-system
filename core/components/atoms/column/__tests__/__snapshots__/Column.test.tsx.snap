// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Column component should match the snapshot with columns 1`] = `
<div
  className="Col--12"
>
  Primary
</div>
`;

exports[`Column component should match the snapshot without columns 1`] = `
<div
  className="Col"
>
  Primary
</div>
`;
